### DISCUSSION TITLE
Use existing media sources from Home Assistant in Music Assistant

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#453] [Feature requests and ideas]
Add ability to browse the existing media sources from Home Assistant within Music Assistant. So in fact just a basic translation from the Home Assistant audio media sources such as the Radio Browser to a Music Assistant music provider.

- The Media source must resolve to an URL (or direct media data) which is processable by MA.

**Comment by ik0adr** [comment]
Hi Marcel and all!!
First of all many thanks for your job.
Unfortunately I cannot enjoy: my music files are in my kodi server (different fron the HA machine), and also if I can play them through the media/DLNA menu, my icons under Music Assistant are greyed, and do not change at all. 
Tryed also to mount the directory in fstab with disastrous ends.. for the USB disk the system uses, resolved by installing everything from scratch.
Am I missing something? I supposed that everything available on the machine would be added to MA automatically, but this is not the case, at least here.
Finally, I do not know how to make the Music directory on kodi resolvable to an URL.. 
Thanks a lot again!!

--Federico

ps sorry if this info was on the discord server, but I was unable to find it..

**Reply by marcelveldt** [reply]
Hi Federico, 
Music Assistant can't access the music files on Kodi and probably never will.
Instead Music Assistant needs access to the music itself, either to online streaming providers or direct access to the files.
So either connect/store your music files to your HA machine or mount a network drive. In the future it will be possible to let MA connect to the network location.

**Reply by ik0adr** [reply]
Hi Marcel, and thanks for kind and speedy reply.
I am now copying all the stuff to a different drive, to be made available locally outside of Kodi, and supposed to be connected by the future MA.
Hope to solve in this way, I will let you  know..
Thanks again, have a nice evening

--Federico

**Comment by PedroKTFC** [comment]
> Add ability to browse the existing media sources from Home Assistant within Music Assistant. So in fact just a basic translation from the Home Assistant audio media sources such as the Radio Browser to a Music Assistant music provider.
> 
> * The Media source must resolve to an URL (or direct media data) which is processable by MA.

Just checking I've understood, you're suggesting allowing access to the Radio Browser in the standard HA Media page? I would really really really like this. It's a pain having to search for a radio station every time.
Thanks

**Comment by erkr** [comment]
> Add ability to browse the existing media sources from Home Assistant within Music Assistant. So in fact just a basic translation from the Home Assistant audio media sources *such as the Radio Browser* to a Music Assistant music provider.
> 

There are an incredible number of radio stations offered. So I would suggest not to provide all radio content, but the option to search and create favorites. 
Same as proposed for local media #512

**Reply by marcelveldt** [reply]
That is exactly what I have in mind. That way you can search a radio station and add it to the library/favorites from there
